File: app/src/main/java/com/pietrantuono/podcasts/apis/PodcastEpisode.java
package com.pietrantuono.podcasts.apis;


import com.rometools.rome.feed.synd.SyndEnclosure;

import java.util.Date;
import java.util.List;

public interface PodcastEpisode {

    String getDuration();

    String getAuthor();

    boolean isExplicit();

    String getImageUrl();

    List<String> getKeywords();

    String getSubtitle();

    String getSummary();

    Date getPubDate();

    String getTitle();

    String getDescription();

    List<? extends SyndEnclosure> getEnclosures();
}

File: app/src/main/java/com/pietrantuono/podcasts/apis/PodcastEpisode.kt
package com.pietrantuono.podcasts.apis


import com.rometools.rome.feed.synd.SyndEnclosure
import java.util.*

interface PodcastEpisode {

    var duration: String?

    var author: String?

    var isExplicit: Boolean?

    var imageUrl: String?

    var keywords: List<String>?

    var subtitle: String?

    var summary: String?

    var pubDate: Date?

    var title: String?

    var description: String?

    var enclosures: List<SyndEnclosure>?

    var downloaded: Boolean?

}